redux/viewMessageDialog.js

```diff
@@ -0,0 +1,31 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+const reducer = {
+	viewMessageDialogOpen(state, action) {
+		return {
+			...state,
+			isOpen: true,
+			message: action.payload,
+		}
+	},
+	viewMessageDialogClosed(state, action) {
+		return {
+			...state,
+			isOpen: false,
+		}
+	},
+}
+
+const viewMessageDialogSlice = createSlice({
+	name: 'ViewMessageDialog',
+	initialState: {
+		isOpen: false,
+	},
+	reducers: reducer,
+})
+
+export const viewMessageDialogReducer = viewMessageDialogSlice.reducer
+export const {
+	viewMessageDialogOpen,
+	viewMessageDialogClosed,
+} = viewMessageDialogSlice.actions
```

Comment:
Look into how redux toolkit recommends you update state in the slices. You no longer need to spread in updated state because RTK (redux toolkit) has built in `immer` which allows you to do the below syntax. I also would recommend not breaking your reducer out and just have it contained within your slice
```suggestion
import { createSlice } from '@reduxjs/toolkit'

const viewMessageDialogSlice = createSlice({
	name: 'ViewMessageDialog',
	initialState: {
		isOpen: false,
	},
	reducers: {
	viewMessageDialogOpen: (state, action) {
	  state.isOpen = true;
	  state.message = action.payload;
	}
     }
})

export const viewMessageDialogReducer = viewMessageDialogSlice.reducer
export const {
	viewMessageDialogOpen,
	viewMessageDialogClosed,
} = viewMessageDialogSlice.actions
```